lib/ecto/adapters/clickhouse/connection.ex

```diff
@@ -914,9 +922,35 @@ defmodule Ecto.Adapters.ClickHouse.Connection do
       message: "unknown or ambiguous (for ClickHouse) Ecto type #{inspect(type)}"
   end
 
-  defp param_type_at(params, ix) do
-    value = Enum.at(params, ix)
-    param_type(value)
+  defp interpolate_param(nil), do: "NULL"
+  defp interpolate_param(true), do: "true"
+  defp interpolate_param(false), do: "false"
+  defp interpolate_param(s) when is_binary(s), do: [?', escape_string(s), ?']
+  defp interpolate_param(i) when is_integer(i), do: Integer.to_string(i)
+  # TODO ensure not scientific notation, or ensure ClickHouse can read it
+  defp interpolate_param(f) when is_float(f), do: Float.to_string(f)
+  defp interpolate_param(%NaiveDateTime{} = naive), do: [?', NaiveDateTime.to_string(naive), ?']
+  # TODO DateTime64
+  defp interpolate_param(%DateTime{} = dt), do: Integer.to_string(DateTime.to_unix(dt, :second))
```

Comment:
Maybe `toDateTime(unix, 'UTC')` to ensure it's treated as a DateTime

Reply:
100%

Reply:
I decided to go with PostgreSQL-style casting with `::` since, I think, it makes it easier to identify the params compared to `toDateTime` and less noisy than `CAST`

```elixir
select: fragment("toDate(toTimeZone(?,'Asia/Taipei'))", ^~U[2024-04-13 11:00:06Z]) == "2024-04-13"
```

```sql
SELECT toDate(toTimeZone('2024-04-13 11:00:06'::DateTime('Etc/UTC'),'Asia/Taipei')) = '2024-04-13'
```
vs
```sql
SELECT toDate(toTimeZone(toDateTime('2024-04-13 11:00:06','Etc/UTC'),'Asia/Taipei')) = '2024-04-13'
```

Reply:
I also learned that `toDate(toTimeZone(...,'Asia/Taipei'))` can be replaced with `toDate(...,'Asia/Taipei')`